worker/download.go

```diff
@@ -755,12 +755,9 @@ loop:
 
 			// handle errors
 			if resp.err != nil {
-				// launch overdrive requests
-				for {
-					if req := s.nextRequest(ctx, resps, true); req != nil {
-						s.launch(req)
-					}
-					break
+				// launch replacement request
+				if req := s.nextRequest(ctx, resps, resp.req.overdrive); req != nil {
```

Comment:
What seemed odd to me here is that we always consider a launched job an overdrive job even if it is just a replacement for a non-overdrive request.

Reply:
It depends on the definition of an overdrive request. Both are fine I think, this is probably better though. It's likely introduced by a small refactor at some point and overlooked until now.